### ISSUE TITLE
Guarantee* unique substitutions for different variable keys

### ISSUE THREAD

**Issue by Splagoon** [Milk-Enterprises/Gambot#28]
Ideally, `$var[a]` and `$var[b]` should be guaranteed to be different values of `$var`.

*provided (# of variables used) <= (# of possible values)

**Comment by emilh91** [comment]
Isn't this done?

**Comment by RobKellett** [comment]
No, right now we can re-use previous substitutions, but $var[1] and $var[2] still have a chance to substitute the same thing.
